add more info, fix database_url

LuceusXylian opened the pull request:
Prints the config path on startup.
Fixes the database_url. The old solution did not worked for me.
Also it will print the database_url on error.

LuceusXylian committed:
add more info, fix database_url

LoipesMas commented:
Thanks for the PR!
Not sure why the old method didn't work. Perhaps because it required `host` to be just hostname (without port). Did you have port in your `db_host`?  
Also this way it wouldn't work for Unix socket paths. It would have to be percent encoded.  
We could do that, but if the issue is what I suspect, it might be better to just add port to config.

LuceusXylian committed:
fixed SQL statements, added user_id

LuceusXylian committed:
added db_port

LuceusXylian commented:
You were right.. I had to split "host:port" to host and port. I added port to the config.

Also I added the DB field user_id. This is actually how it should be. The user_id allows us to change the username at will without issues and the user_id helps to integrate this project to other projects.

LuceusXylian committed:
fix UserJoined message

LoipesMas commented:
Those are some great improvements, thanks!
Moving allow/deny list to the database is probably the right move. It started with me wanting to have operators set from config, but I probably should just add commandline to the server and have operators set from there.

PS: Note to self: remember to fix error handling that you leave for later.
